sparkle/src/components/DropdownMenu.tsx

```diff
@@ -102,135 +103,165 @@ export interface DropdownButtonProps {
   className?: string;
   disabled?: boolean;
   children?: React.ReactNode;
+  ref?: React.Ref<HTMLButtonElement>;
+  onClick?: () => void;
 }
 
-DropdownMenu.Button = function ({
-  label,
-  type = "menu",
-  size = "sm",
-  tooltip,
-  icon,
-  children,
-  tooltipPosition = "above",
-  className = "",
-  disabled = false,
-}: DropdownButtonProps) {
-  const finalLabelClasses = classNames(
-    labelClasses.base,
-    labelSizeClasses[size],
-    labelClasses.dark.base,
-    !disabled ? labelClasses.active : "",
-    !disabled ? labelClasses.dark.active : "",
-    !disabled ? labelClasses.hover : "",
-    !disabled ? labelClasses.dark.hover : "",
-    disabled ? labelClasses.disabled : ""
-  );
+DropdownMenu.Button = forwardRef<HTMLButtonElement, DropdownButtonProps>(
+  function DropdownMenuButton(
+    {
+      label,
+      type = "menu",
+      size = "sm",
+      tooltip,
+      icon,
+      children,
+      tooltipPosition = "above",
+      className = "",
+      disabled = false,
+      onClick,
+    },
+    forwardedRef
+  ) {
+    const contextRef = useContext(ButtonRefContext);
+    const internalRef = useRef<HTMLButtonElement | null>(null);
+
+    const finalLabelClasses = classNames(
+      labelClasses.base,
+      labelSizeClasses[size],
+      labelClasses.dark.base,
+      !disabled ? labelClasses.active : "",
+      !disabled ? labelClasses.dark.active : "",
+      !disabled ? labelClasses.hover : "",
+      !disabled ? labelClasses.dark.hover : "",
+      disabled ? labelClasses.disabled : ""
+    );
 
-  const finalIconClasses = classNames(
-    iconClasses.base,
-    iconClasses.dark.base,
-    !disabled ? iconClasses.hover : "",
-    !disabled ? iconClasses.dark.hover : "",
-    disabled ? iconClasses.disabled : ""
-  );
+    const finalIconClasses = classNames(
+      iconClasses.base,
+      iconClasses.dark.base,
+      !disabled ? iconClasses.hover : "",
+      !disabled ? iconClasses.dark.hover : "",
+      disabled ? iconClasses.disabled : ""
+    );
 
-  const finalChevronClasses = classNames(
-    chevronClasses.base,
-    !disabled ? chevronClasses.hover : "",
-    chevronClasses.dark.base,
-    !disabled ? chevronClasses.dark.hover : "",
-    disabled ? chevronClasses.disabled : ""
-  );
+    const finalChevronClasses = classNames(
+      chevronClasses.base,
+      !disabled ? chevronClasses.hover : "",
+      chevronClasses.dark.base,
+      !disabled ? chevronClasses.dark.hover : "",
+      disabled ? chevronClasses.disabled : ""
+    );
 
-  const buttonRef = useContext(ButtonRefContext);
+    useEffect(() => {
+      if (contextRef) {
+        contextRef.current = internalRef.current;
+      }
+      if (typeof forwardedRef === "function") {
+        forwardedRef(internalRef.current);
+      } else if (forwardedRef) {
+        forwardedRef.current = internalRef.current;
+      }
+    }, [contextRef, forwardedRef]);
```

Comment:
instead of an effect, define a new ref function : 
```suggestion
    const aggregatedRef = (value: HTMLButtonElement) => {
      if (contextRef) {
        contextRef.current = value;
      }
      if (typeof forwardedRef === "function") {
        forwardedRef(value);
      } else if (forwardedRef) {
        forwardedRef.current = value;
      }
    };
```

and then use ref={aggregatedRef} in Menu.Button